diff --git a/linker/linker.c b/linker/linker.c
index e0a8a18..c4ceea4 100644
--- a/linker/linker.c
+++ b/linker/linker.c
@@ -782,7 +782,7 @@ get_lib_extents(int fd, const char *name, void *__hdr, unsigned *total_sz)
     min_vaddr &= ~PAGE_MASK;
 
     /* round max_vaddr up to the next page */
-    max_vaddr = (max_vaddr + PAGE_SIZE - 1) & ~PAGE_MASK;
+    max_vaddr = (max_vaddr + PAGE_SIZE * 2 - 1) & ~PAGE_MASK;
 
     *total_sz = (max_vaddr - min_vaddr);
     return (unsigned)req_base;
