diff --git a/fonts.mk b/fonts.mk
index 8ea4fd3..3ccdafb 100644
--- a/fonts.mk
+++ b/fonts.mk
@@ -22,6 +22,7 @@ PRODUCT_COPY_FILES := \
     frameworks/base/data/fonts/DroidSansThai.ttf:system/fonts/DroidSansThai.ttf \
     frameworks/base/data/fonts/DroidSansArmenian.ttf:system/fonts/DroidSansArmenian.ttf \
     frameworks/base/data/fonts/DroidSansGeorgian.ttf:system/fonts/DroidSansGeorgian.ttf \
+    frameworks/base/data/fonts/DroidSansFallback.ttf:system/fonts/DroidSansFallback.ttf \
 
 # MozTT fonts
 PRODUCT_COPY_FILES := \
