diff --git a/content/media/omx/OmxDecoder.cpp b/content/media/omx/OmxDecoder.cpp
index 2fdfa7e..895b6f3 100644
--- a/content/media/omx/OmxDecoder.cpp
+++ b/content/media/omx/OmxDecoder.cpp
@@ -245,7 +245,7 @@ bool OmxDecoder::Init() {
     // is currently used, and for formats this code is currently used
     // for (h.264).  So if we don't get a hardware decoder, just give
     // up.
-    int flags = kHardwareCodecsOnly;
+    int flags = kSoftwareCodecsOnly;//kHardwareCodecsOnly;
     videoSource = OMXCodec::Create(omx,
                                    videoTrack->getFormat(),
                                    false, // decoder
