diff --git a/sp7702/prod_sp7702cuccspecAplus.mk b/sp7702/prod_sp7702cuccspecAplus.mk
index eabb007..023edf2 100644
--- a/sp7702/prod_sp7702cuccspecAplus.mk
+++ b/sp7702/prod_sp7702cuccspecAplus.mk
@@ -82,7 +82,7 @@ $(call inherit-product, vendor/sprd/operator/cucc/specA/res/apn/apn_res.mk)
 $(call inherit-product, $(BOARDDIR)/base.mk)
 $(call inherit-product, $(BOARDDIR)/proprietories.mk)
 $(call inherit-product, vendor/sprd/operator/cucc/specA/res/boot/boot_res.mk)
-$(call inherit-product, vendor/sprd/UniverseUI/universeui.mk)
+#$(call inherit-product, vendor/sprd/UniverseUI/universeui.mk)
 
 # include standard configs
 $(call inherit-product, $(SRC_TARGET_DIR)/product/full_base.mk)
diff --git a/sp7702/prod_sp7702tri.mk b/sp7702/prod_sp7702tri.mk
index 2677dfd..5e1af45 100644
--- a/sp7702/prod_sp7702tri.mk
+++ b/sp7702/prod_sp7702tri.mk
@@ -77,7 +77,7 @@ $(call inherit-product, frameworks/native/build/phone-hdpi-256-dalvik-heap.mk)
 $(call inherit-product, $(BOARDDIR)/base.mk)
 $(call inherit-product, $(BOARDDIR)/proprietories.mk)
 $(call inherit-product, device/sprd/common/res/boot/boot_res.mk)
-$(call inherit-product, vendor/sprd/UniverseUI/universeui.mk)
+#$(call inherit-product, vendor/sprd/UniverseUI/universeui.mk)
 
 # include standard configs
 $(call inherit-product, $(SRC_TARGET_DIR)/product/full_base.mk)
diff --git a/sp7710ga/prod_sp7710gacuccspecAplus.mk b/sp7710ga/prod_sp7710gacuccspecAplus.mk
index f6adcd3..c43e64c 100644
--- a/sp7710ga/prod_sp7710gacuccspecAplus.mk
+++ b/sp7710ga/prod_sp7710gacuccspecAplus.mk
@@ -79,7 +79,7 @@ $(call inherit-product, vendor/sprd/operator/cucc/specA/res/apn/apn_res.mk)
 $(call inherit-product, $(BOARDDIR)/base.mk)
 $(call inherit-product, $(BOARDDIR)/proprietories.mk)
 $(call inherit-product, vendor/sprd/operator/cucc/specA/res/boot/boot_res.mk)
-$(call inherit-product, vendor/sprd/UniverseUI/universeui.mk)
+#$(call inherit-product, vendor/sprd/UniverseUI/universeui.mk)
 
 # include standard configs
 $(call inherit-product, $(SRC_TARGET_DIR)/product/full_base.mk)
diff --git a/sp7710ga/prod_sp7710gatri.mk b/sp7710ga/prod_sp7710gatri.mk
index 25d3dcd..6403a42 100644
--- a/sp7710ga/prod_sp7710gatri.mk
+++ b/sp7710ga/prod_sp7710gatri.mk
@@ -73,7 +73,7 @@ $(call inherit-product, frameworks/native/build/phone-hdpi-dalvik-heap.mk)
 $(call inherit-product, $(BOARDDIR)/base.mk)
 $(call inherit-product, $(BOARDDIR)/proprietories.mk)
 $(call inherit-product, vendor/sprd/operator/cucc/specA/res/boot/boot_res.mk)
-$(call inherit-product, vendor/sprd/UniverseUI/universeui.mk)
+#$(call inherit-product, vendor/sprd/UniverseUI/universeui.mk)
 
 # include standard configs
 $(call inherit-product, $(SRC_TARGET_DIR)/product/full_base.mk)
diff --git a/sp8810ea/prod_sp8810cmccspecAplus_UUI.mk b/sp8810ea/prod_sp8810cmccspecAplus_UUI.mk
index c6a2dfb..2e2e243 100644
--- a/sp8810ea/prod_sp8810cmccspecAplus_UUI.mk
+++ b/sp8810ea/prod_sp8810cmccspecAplus_UUI.mk
@@ -86,7 +86,7 @@ $(call inherit-product, frameworks/native/build/phone-hdpi-dalvik-heap.mk)
 $(call inherit-product, $(BOARDDIR)/base.mk)
 $(call inherit-product, $(BOARDDIR)/proprietories.mk)
 $(call inherit-product, device/sprd/common/res/boot/boot_res.mk)
-$(call inherit-product, vendor/sprd/UniverseUI/universeui.mk)
+#$(call inherit-product, vendor/sprd/UniverseUI/universeui.mk)
 
 # include standard configs
 $(call inherit-product, $(SRC_TARGET_DIR)/product/full_base.mk)
diff --git a/sp8825ea/AndroidProducts.mk b/sp8825ea/AndroidProducts.mk
index cbce02a..f4a95cf 100644
--- a/sp8825ea/AndroidProducts.mk
+++ b/sp8825ea/AndroidProducts.mk
@@ -20,4 +20,4 @@ PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8825eacmccspecAplus.mk
 
 PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8825eanativebase.mk
 PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8825eanativeplus.mk
-PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8825eacmccspecAplus_UUI.mk
+#PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8825eacmccspecAplus_UUI.mk
diff --git a/sp8825ea/BoardConfig.mk b/sp8825ea/BoardConfig.mk
index b2563de..6ae89a6 100644
--- a/sp8825ea/BoardConfig.mk
+++ b/sp8825ea/BoardConfig.mk
@@ -82,7 +82,7 @@ BOARD_WPA_SUPPLICANT_PRIVATE_LIB := lib_driver_cmd_bcmdhd
 BOARD_HOSTAPD_DRIVER        := NL80211
 BOARD_HOSTAPD_PRIVATE_LIB   := lib_driver_cmd_bcmdhd
 BOARD_WLAN_DEVICE           := bcmdhd
-HAVE_WLAN_CMCC_FEATURE      := true
+HAVE_WLAN_CMCC_FEATURE      := false
 WIFI_DRIVER_FW_PATH_PARAM   := "/sys/module/bcmdhd/parameters/firmware_path"
 WIFI_DRIVER_FW_PATH_STA     := "/etc/wifi/fw_bcmdhd.bin"
 WIFI_DRIVER_FW_PATH_P2P     := "/etc/wifi/fw_bcmdhd_p2p.bin"
diff --git a/sp8825ea/base.mk b/sp8825ea/base.mk
old mode 100755
new mode 100644
diff --git a/sp8825ea/init.rc b/sp8825ea/init.rc
index 5eb61cb..8cdbb55 100644
--- a/sp8825ea/init.rc
+++ b/sp8825ea/init.rc
@@ -16,6 +16,8 @@ on early-init
 
 # create mountpoints
     mkdir /mnt 0775 root system
+    
+    import /init.b2g.rc
 
 on init
 
@@ -122,6 +124,9 @@ on fs
     #mount yaffs2 mtd@userdata /data nosuid nodev
     #mount yaffs2 mtd@cache /cache nosuid nodev
     mkdir /productinfo 0777 system system
+    chmod 0755 /system/b2g/b2g
+    chmod 0755 /system/b2g/updater
+    chmod 0755 /system/b2g/plugin-container
 # ---------------------------- spreadtrum on fs section end
 
 on post-fs
@@ -370,6 +375,10 @@ service console /system/bin/sh
 
 on property:ro.debuggable=1
     start console
+# add core dump
+    setrlimit 4 -1 -1
+    write /proc/sys/kernel/core_pattern /data/tombstones/core.%e
+    write /proc/sys/fs/suid_dumpable 1
 
 # adbd is controlled via property triggers in init.<platform>.usb.rc
 service adbd /sbin/adbd
@@ -397,7 +406,7 @@ service servicemanager /system/bin/servicemanager
     user system
     group system
     critical
-    onrestart restart zygote
+#    onrestart restart zygote
     onrestart restart media
     onrestart restart surfaceflinger
     onrestart restart drm
@@ -427,15 +436,15 @@ service surfaceflinger /system/bin/surfaceflinger
     class main
     user system
     group graphics
-    onrestart restart zygote
+#    onrestart restart zygote
 
-service zygote /system/bin/app_process -Xzygote /system/bin --zygote --start-system-server
-    class main
-    socket zygote stream 660 root system
-    onrestart write /sys/android_power/request_state wake
-    onrestart write /sys/power/state on
-    onrestart restart media
-    onrestart restart netd
+#service zygote /system/bin/app_process -Xzygote /system/bin --zygote --start-system-server
+#    class main
+#    socket zygote stream 660 root system
+#    onrestart write /sys/android_power/request_state wake
+#    onrestart write /sys/power/state on
+#    onrestart restart media
+#   onrestart restart netd
 
 service drm /system/bin/drmserver
     class main
diff --git a/sp8825ea/init.sc8825.rc b/sp8825ea/init.sc8825.rc
old mode 100755
new mode 100644
index e5597ba..959c4c0
--- a/sp8825ea/init.sc8825.rc
+++ b/sp8825ea/init.sc8825.rc
@@ -47,7 +47,8 @@ on boot
     chown bluetooth bluetooth /sys/class/rfkill/rfkill0/type
     chown bluetooth system /proc/bluetooth/sleep/proto
     mkdir /data/misc/wifi/sockets 0770 wifi wifi
-    mkdir /data/system 0770 system system
+    #for b2g brcm wifi permission, mkdir /data/system/wpa_supplicant 
+    mkdir /data/system 0771 system system
     mkdir /etc/wifi 0770 wifi wifi
     mkdir /data/system/wpa_supplicant 0771 wifi wifi
     mkdir /data/misc/dhcp 0770 dhcp dhcp
diff --git a/sp8825ea/prod_sp8825eacmccspecAplus_UUI.mk b/sp8825ea/prod_sp8825eacmccspecAplus_UUI.mk
index 6b0c46f..f4c3e14 100644
--- a/sp8825ea/prod_sp8825eacmccspecAplus_UUI.mk
+++ b/sp8825ea/prod_sp8825eacmccspecAplus_UUI.mk
@@ -87,7 +87,7 @@ $(call inherit-product, frameworks/native/build/phone-hdpi-dalvik-heap.mk)
 $(call inherit-product, $(BOARDDIR)/base.mk)
 $(call inherit-product, $(BOARDDIR)/proprietories.mk)
 $(call inherit-product, device/sprd/common/res/boot/boot_res.mk)
-$(call inherit-product, vendor/sprd/UniverseUI/universeui.mk)
+#$(call inherit-product, vendor/sprd/UniverseUI/universeui.mk)
 
 # include standard configs
 $(call inherit-product, $(SRC_TARGET_DIR)/product/full_base.mk)
