diff --git a/apps/system/js/statusbar.js b/apps/system/js/statusbar.js
index af01697..c49e8b4 100644
--- a/apps/system/js/statusbar.js
+++ b/apps/system/js/statusbar.js
@@ -15,19 +15,29 @@ function AnimatedIcon(element, path, frames, delay) {
   this.frame = 1;
   this.frames = frames;
   this.timerId = null;
+  this._started = false;
+  var image;
 
   // Load the image and paint the first frame
-  var image = new Image();
-  image.src = path;
-  image.onload = function() {
-    var w = image.width;
-    var h = image.height / frames;
-
-    context.drawImage(image, 0, 0, w, h, 0, 0, w, h);
-  };
+  function init() {
+    image = new Image();
+    image.src = path;
+    image.onload = function() {
+      var w = image.width;
+      var h = image.height / frames;
+      context.drawImage(image, 0, 0, w, h, 0, 0, w, h);
+    };
+  }
 
   this.start = function() {
     var self = this;
+    // XXX: If we draw canvas during device start up,
+    // it will face following issue.
+    // https://bugzilla.mozilla.org/show_bug.cgi?id=849736
+    if (!self._started) {
+      self._started = true;
+      init();
+    }
 
     if (this.timerId == null) {
       this.timerId = setInterval(function() {
@@ -174,9 +184,9 @@ var StatusBar = {
     // Create the objects used to animate the statusbar-network-activity and
     // statusbar-system-downloads canvas elements
     this.networkActivityAnimation = new AnimatedIcon(this.icons.networkActivity,
-      'style/statusbar/images/network-activity-flat.png', 6, 200);
+      '/style/statusbar/images/network-activity-flat.png', 6, 200);
     this.systemDownloadsAnimation = new AnimatedIcon(this.icons.systemDownloads,
-      'style/statusbar/images/system-downloads-flat.png', 8, 130);
+      '/style/statusbar/images/system-downloads-flat.png', 8, 130);
 
     this.setActive(true);
   },
