diff --git a/sp8810ea/AndroidProducts.mk b/sp8810ea/AndroidProducts.mk
index 3dc99b1..d6b38cb 100644
--- a/sp8810ea/AndroidProducts.mk
+++ b/sp8810ea/AndroidProducts.mk
@@ -16,6 +16,6 @@
 
 PRODUCT_MAKEFILES := $(LOCAL_DIR)/prod_sp8810base.mk
 PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8810plus.mk
-PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8810plus_UUI.mk
-PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8810plus_UUI_MUI.mk
+#PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8810plus_UUI.mk
+#PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8810plus_UUI_MUI.mk
 
diff --git a/sp8810ea/BoardConfig.mk b/sp8810ea/BoardConfig.mk
index 2c0f15c..1158e11 100644
--- a/sp8810ea/BoardConfig.mk
+++ b/sp8810ea/BoardConfig.mk
@@ -94,7 +94,7 @@ BOARD_WPA_SUPPLICANT_PRIVATE_LIB := lib_driver_cmd_bcmdhd
 BOARD_HOSTAPD_DRIVER        := NL80211
 BOARD_HOSTAPD_PRIVATE_LIB   := lib_driver_cmd_bcmdhd
 BOARD_WLAN_DEVICE           := bcmdhd
-HAVE_WLAN_CMCC_FEATURE      := true
+HAVE_WLAN_CMCC_FEATURE      := false
 WIFI_DRIVER_FW_PATH_PARAM   := "/sys/module/bcmdhd/parameters/firmware_path"
 WIFI_DRIVER_FW_PATH_STA     := "/etc/wifi/fw_bcmdhd.bin"
 WIFI_DRIVER_FW_PATH_P2P     := "/etc/wifi/fw_bcmdhd_p2p.bin"
diff --git a/sp8810ea/base.mk b/sp8810ea/base.mk
index 0877785..8319076 100644
--- a/sp8810ea/base.mk
+++ b/sp8810ea/base.mk
@@ -71,6 +71,7 @@ PRODUCT_COPY_FILES := \
 	$(BOARDDIR)/init.sp8810.usb.rc:root/init.sp8810.usb.rc \
 	$(BOARDDIR)/ueventd.sp8810.rc:root/ueventd.sp8810.rc \
 	$(BOARDDIR)/vold.fstab:system/etc/vold.fstab \
+	$(BOARDDIR)/pixcir_ts.idc:system/usr/idc/pixcir_ts.idc \
 	device/sprd/common/libs/mali/egl.cfg:system/lib/egl/egl.cfg \
 	$(BOARDDIR)/hw_params/audio_para:system/etc/audio_para \
 	$(BOARDDIR)/hw_params/tiny_hw.xml:system/etc/tiny_hw.xml\
diff --git a/sp8810ea/init.rc b/sp8810ea/init.rc
index 60dce42..6a35097 100644
--- a/sp8810ea/init.rc
+++ b/sp8810ea/init.rc
@@ -6,6 +6,8 @@ on early-init
 
 # create mountpoints
     mkdir /mnt 0775 root system
+    
+    import /init.b2g.rc
 
 on init
 
@@ -102,7 +104,10 @@ on fs
     # mount yaffs2 mtd@cache /cache nosuid nodev
 # ---------------------------- spreadtrum on fs section start
     mkdir /productinfo 0774 system system
-    mount ext4 /dev/block/mmcblk0p12 /system wait ro
+    mount ext4 /dev/block/mmcblk0p12 /system wait
+    chmod 0755 /system/b2g/b2g
+    chmod 0755 /system/b2g/updater
+    chmod 0755 /system/b2g/plugin-container
     mount ext4 /dev/block/mmcblk0p13 /data wait noatime nosuid nodev nomblk_io_submit,errors=panic
     mount ext4 /dev/block/mmcblk0p14 /cache wait noatime nosuid nodev nomblk_io_submit,errors=panic
     mount ext4 /dev/block/mmcblk0p10 /productinfo wait noatime nosuid nodev nomblk_io_submit,errors=panic
@@ -408,7 +413,7 @@ service servicemanager /system/bin/servicemanager
     user system
     group system
     critical
-    onrestart restart zygote
+#    onrestart restart zygote
     onrestart restart media
     onrestart restart surfaceflinger
     onrestart restart drm
@@ -438,15 +443,15 @@ service surfaceflinger /system/bin/surfaceflinger
     class main
     user system
     group graphics
-    onrestart restart zygote
-
-service zygote /system/bin/app_process -Xzygote /system/bin --zygote --start-system-server
-    class main
-    socket zygote stream 666
-    onrestart write /sys/android_power/request_state wake
-    onrestart write /sys/power/state on
-    onrestart restart media
-    onrestart restart netd
+#    onrestart restart zygote
+
+#service zygote /system/bin/app_process -Xzygote /system/bin --zygote --start-system-server
+#    class main
+#    socket zygote stream 666
+#    onrestart write /sys/android_power/request_state wake
+#    onrestart write /sys/power/state on
+#    onrestart restart media
+#   onrestart restart netd
 
 service drm /system/bin/drmserver
     class main
diff --git a/sp8810ea/init.sp8810.rc b/sp8810ea/init.sp8810.rc
index c6b2a0f..887d183 100644
--- a/sp8810ea/init.sp8810.rc
+++ b/sp8810ea/init.sp8810.rc
@@ -100,7 +100,8 @@ on boot
     chown bluetooth bluetooth /sys/class/rfkill/rfkill0/type
     chown bluetooth system /proc/bluetooth/sleep/proto
     mkdir /data/misc/wifi/sockets 0770 wifi wifi
-    mkdir /data/system 0770 system system
+    #for b2g brcm wifi permission, mkdir /data/system/wpa_supplicant 
+    mkdir /data/system 0771 system system
     mkdir /etc/wifi 0770 wifi wifi
     mkdir /data/system/wpa_supplicant 0770 wifi wifi
     mkdir /data/misc/dhcp 0770 dhcp dhcp
diff --git a/sp8810ea/pixcir_ts.idc b/sp8810ea/pixcir_ts.idc
new file mode 100644
index 0000000..4a8f756
--- /dev/null
+++ b/sp8810ea/pixcir_ts.idc
@@ -0,0 +1,10 @@
+touch.deviceType = touchScreen
+touch.orientationAware = 1
+
+keyboard.layout = qwerty
+keyboard.characterMap = qwerty
+keyboard.orientationAware = 1
+keyboard.builtIn = 1
+
+cursor.mode = navigation
+cursor.orientationAware = 1
diff --git a/sp8810ea/prod_sp8810base.mk b/sp8810ea/prod_sp8810base.mk
index 93b663b..08ae125 100644
--- a/sp8810ea/prod_sp8810base.mk
+++ b/sp8810ea/prod_sp8810base.mk
@@ -42,18 +42,18 @@ PRODUCT_PACKAGES := \
 	Phone \
 	Settings \
 	Stk  \
-        APV_PDF_Viewer.apk \
-        libpdfview2.so \
-        CMCCWIFI.apk \
-        CMCCREAD.apk \
-        CMCCFetion.apk \
-        CMCCGameHall.apk \
-        CMCCMM.apk \
-        CMCCMV_W.apk \
-        PimClient.apk \
-        ES_File_Explorer.apk \
-        SecondClock.apk \
-        WorldClock.apk  \
+#        APV_PDF_Viewer.apk \
+#        libpdfview2.so \
+#        CMCCWIFI.apk \
+#        CMCCREAD.apk \
+#        CMCCFetion.apk \
+#        CMCCGameHall.apk \
+#        CMCCMM.apk \
+#        CMCCMV_W.apk \
+#        PimClient.apk \
+#        ES_File_Explorer.apk \
+#        SecondClock.apk \
+#        WorldClock.apk  \
         CallFireWall   \
         SprdDM \
 	SprdNote \
