Based on jb-b2g revision 316979186d5b91fdfc450b0d709cb80a1ece2b78 commit
diff --git a/core/disabled_modules.mk b/core/disabled_modules.mk
index f55273d..26caa38 100644
--- a/core/disabled_modules.mk
+++ b/core/disabled_modules.mk
@@ -18,7 +18,6 @@ DISABLED_USER_MODULES := \
 	jdwpspy \
 	a3dconvert \
 	libsampleplugin \
-	libstagefright_chromium_http \
 	libjavacore \
 	app_process__asan \
 
diff --git a/target/product/full_base.mk b/target/product/full_base.mk
index 6a702da..10918da 100644
--- a/target/product/full_base.mk
+++ b/target/product/full_base.mk
@@ -62,3 +62,6 @@ $(call inherit-product, $(SRC_TARGET_DIR)/product/locales_full.mk)
 
 # Get everything else from the parent package
 $(call inherit-product, $(SRC_TARGET_DIR)/product/generic_no_telephony.mk)
+
+# B2G
+$(call inherit-product, gonk-misc/b2g.mk)
