From commit 2ed11a786d15f90ff846e6113432a70778238d8e
diff --git a/common/tools/slog/Android.mk b/common/tools/slog/Android.mk
index 35531f1..bbbb353 100644
--- a/common/tools/slog/Android.mk
+++ b/common/tools/slog/Android.mk
@@ -6,9 +6,9 @@ LOCAL_SRC_FILES := slog.c handler.c parse_conf.c screenshot.c
 LOCAL_MODULE := slog
 LOCAL_STATIC_LIBRARIES := libcutils libc
 LOCAL_MODULE_TAGS := optional
-ifeq ($(PLATFORM_VERSION),4.1.2)
+#ifeq ($(PLATFORM_VERSION),4.1.2)
 LOCAL_CFLAGS += -DSLOG_ALOGD_ALOGE
-endif
+#endif
 ifeq ($(PLATFORM_VERSION),2.3.5)
 LOCAL_CFLAGS += -DSLOG_BTLOG_235
 endif
@@ -23,9 +23,9 @@ LOCAL_SRC_FILES := slogctl.c
 LOCAL_MODULE := slogctl
 LOCAL_STATIC_LIBRARIES := libcutils libc
 LOCAL_MODULE_TAGS := optional
-ifeq ($(PLATFORM_VERSION),4.1.2)
+#ifeq ($(PLATFORM_VERSION),4.1.2)
 LOCAL_CFLAGS += -DSLOG_ALOGD_ALOGE
-endif
+#endif
 LOCAL_LDLIBS += -lpthread
 LOCAL_C_INCLUDES += external/jpeg external/zlib
 LOCAL_SHARED_LIBRARIES := liblog libz libjpeg
diff --git a/sp7702/AndroidProducts.mk b/sp7702/AndroidProducts.mk
index 726efaa..9fb3579 100644
--- a/sp7702/AndroidProducts.mk
+++ b/sp7702/AndroidProducts.mk
@@ -17,5 +17,5 @@
 PRODUCT_MAKEFILES := $(LOCAL_DIR)/prod_sp7702base.mk
 PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp7702plus.mk
 PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp7702tri.mk
-PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp7702cuccspecAplus.mk
+#PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp7702cuccspecAplus.mk
 
diff --git a/sp7702/prod_sp7702base.mk b/sp7702/prod_sp7702base.mk
index 3bbf861..6c891a5 100644
--- a/sp7702/prod_sp7702base.mk
+++ b/sp7702/prod_sp7702base.mk
@@ -58,16 +58,16 @@ PRODUCT_PACKAGES += \
 PRODUCT_PACKAGES += \
 	
 # add  system properties
-PRODUCT_PROPERTY_OVERRIDES += \
-	ro.operator=cmcc \
-	ro.operator.version=specA
+#PRODUCT_PROPERTY_OVERRIDES += \
+#	ro.operator=cmcc \
+#	ro.operator.version=specA
 	
 PRODUCT_COPY_FILES := \
 	$(BOARDDIR)/sprd-keypad.kl:system/usr/keylayout/sprd-keypad.kl \
 	$(BOARDDIR)/pixcir_ts.kl:system/usr/keylayout/pixcir_ts.kl \
 	$(BOARDDIR)/pixcir_ts.idc:system/usr/idc/pixcir_ts.idc
 
-$(call inherit-product, frameworks/native/build/phone-hdpi-256-dalvik-heap.mk)
+#$(call inherit-product, frameworks/native/build/phone-hdpi-256-dalvik-heap.mk)
 
 # include classified configs
 $(call inherit-product, $(BOARDDIR)/base.mk)
diff --git a/sp7702/prod_sp7702cuccspecAplus.mk b/sp7702/prod_sp7702cuccspecAplus.mk
index 6ee8792..4c5eef7 100644
--- a/sp7702/prod_sp7702cuccspecAplus.mk
+++ b/sp7702/prod_sp7702cuccspecAplus.mk
@@ -82,7 +82,7 @@ $(call inherit-product, vendor/sprd/operator/cucc/specA/res/apn/apn_res.mk)
 $(call inherit-product, $(BOARDDIR)/base.mk)
 $(call inherit-product, $(BOARDDIR)/proprietories.mk)
 $(call inherit-product, vendor/sprd/operator/cucc/specA/res/boot/boot_res.mk)
-$(call inherit-product, vendor/sprd/UniverseUI/universeui.mk)
+#$(call inherit-product, vendor/sprd/UniverseUI/universeui.mk)
 
 # include standard configs
 $(call inherit-product, $(SRC_TARGET_DIR)/product/full_base.mk)
diff --git a/sp7702/prod_sp7702plus.mk b/sp7702/prod_sp7702plus.mk
index 1e4840a..09ccbcc 100644
--- a/sp7702/prod_sp7702plus.mk
+++ b/sp7702/prod_sp7702plus.mk
@@ -60,16 +60,16 @@ PRODUCT_PACKAGES += \
 PRODUCT_PACKAGES += \
 	
 # add  system properties
-PRODUCT_PROPERTY_OVERRIDES += \
-	ro.operator=cmcc \
-	ro.operator.version=specA
+#PRODUCT_PROPERTY_OVERRIDES += \
+#	ro.operator=cmcc \
+#	ro.operator.version=specA
 	
 PRODUCT_COPY_FILES := \
 	$(BOARDDIR)/sprd-keypad.kl:system/usr/keylayout/sprd-keypad.kl \
 	$(BOARDDIR)/pixcir_ts.kl:system/usr/keylayout/pixcir_ts.kl \
 	$(BOARDDIR)/pixcir_ts.idc:system/usr/idc/pixcir_ts.idc
 
-$(call inherit-product, frameworks/native/build/phone-hdpi-256-dalvik-heap.mk)
+#$(call inherit-product, frameworks/native/build/phone-hdpi-256-dalvik-heap.mk)
 
 # include classified configs
 $(call inherit-product, $(BOARDDIR)/base.mk)
diff --git a/sp7702/prod_sp7702tri.mk b/sp7702/prod_sp7702tri.mk
index 57825e6..e6e3b86 100644
--- a/sp7702/prod_sp7702tri.mk
+++ b/sp7702/prod_sp7702tri.mk
@@ -62,22 +62,22 @@ PRODUCT_PACKAGES += \
 PRODUCT_PACKAGES += \
 	
 # add  system properties
-PRODUCT_PROPERTY_OVERRIDES += \
-	ro.operator=cmcc \
-	ro.operator.version=specA
+#PRODUCT_PROPERTY_OVERRIDES += \
+#	ro.operator=cmcc \
+#	ro.operator.version=specA
 	
 PRODUCT_COPY_FILES := \
 	$(BOARDDIR)/sprd-keypad.kl:system/usr/keylayout/sprd-keypad.kl \
 	$(BOARDDIR)/pixcir_ts.kl:system/usr/keylayout/pixcir_ts.kl \
 	$(BOARDDIR)/pixcir_ts.idc:system/usr/idc/pixcir_ts.idc
 
-$(call inherit-product, frameworks/native/build/phone-hdpi-256-dalvik-heap.mk)
+#$(call inherit-product, frameworks/native/build/phone-hdpi-256-dalvik-heap.mk)
 
 # include classified configs
 $(call inherit-product, $(BOARDDIR)/base.mk)
 $(call inherit-product, $(BOARDDIR)/proprietories.mk)
 $(call inherit-product, device/sprd/common/res/boot/boot_res.mk)
-$(call inherit-product, vendor/sprd/UniverseUI/universeui.mk)
+#$(call inherit-product, vendor/sprd/UniverseUI/universeui.mk)
 
 # include standard configs
 $(call inherit-product, $(SRC_TARGET_DIR)/product/full_base.mk)
diff --git a/sp7710ga/AndroidProducts.mk b/sp7710ga/AndroidProducts.mk
index 1442bc3..1bb5492 100644
--- a/sp7710ga/AndroidProducts.mk
+++ b/sp7710ga/AndroidProducts.mk
@@ -17,5 +17,5 @@
 PRODUCT_MAKEFILES := $(LOCAL_DIR)/prod_sp7710ga.mk
 PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp7710gahvga.mk
 PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp7710gatri.mk
-PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp7710gacuccspecAplus.mk
+#PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp7710gacuccspecAplus.mk
 
diff --git a/sp7710ga/base.mk b/sp7710ga/base.mk
index db153b2..19487ee 100644
--- a/sp7710ga/base.mk
+++ b/sp7710ga/base.mk
@@ -46,8 +46,8 @@ PRODUCT_PACKAGES += \
     vtserver
 
 # prebuild files
-PRODUCT_PACKAGES += \
-    ES_File_Explorer.apk
+#PRODUCT_PACKAGES += \
+#    ES_File_Explorer.apk
 
 PRODUCT_PACKAGES += \
 	modem_control\
diff --git a/sp7710ga/init.rc b/sp7710ga/init.rc
index dd3934b..3149ae2 100644
--- a/sp7710ga/init.rc
+++ b/sp7710ga/init.rc
@@ -17,6 +17,8 @@ on early-init
 # create mountpoints
     mkdir /mnt 0775 root system
 
+    import /init.b2g.rc
+
 on init
 
 sysclktz 0
@@ -126,13 +128,16 @@ on fs
     start modem_control
 
     mount yaffs2 mtd@system /system
+    chmod 0755 /system/b2g/b2g
+    chmod 0755 /system/b2g/updater
+    chmod 0755 /system/b2g/plugin-container
     mount yaffs2 mtd@system /system ro remount
     mount yaffs2 mtd@userdata /data nosuid nodev
     mount yaffs2 mtd@cache /cache nosuid nodev
 
 on post-fs
     # once everything is setup, no need to modify /
-    mount rootfs rootfs / ro remount
+#    mount rootfs rootfs / ro remount
 
     mount yaffs2 mtd@fixnv /fixnv nosuid nodev no-checkpoint,tags-ecc-off
     chown system system /fixnv
@@ -397,6 +402,10 @@ service console /system/bin/sh
 
 on property:ro.debuggable=1
     start console
+# add core dump
+    setrlimit 4 -1 -1
+    write /proc/sys/kernel/core_pattern /data/tombstones/core.%e
+    write /proc/sys/fs/suid_dumpable 1
 
 # adbd is controlled via property triggers in init.<platform>.usb.rc
 service adbd /sbin/adbd
@@ -424,7 +433,7 @@ service servicemanager /system/bin/servicemanager
     user system
     group system
     critical
-    onrestart restart zygote
+#    onrestart restart zygote
     onrestart restart media
     onrestart restart surfaceflinger
     onrestart restart drm
@@ -454,15 +463,15 @@ service surfaceflinger /system/bin/surfaceflinger
     class main
     user system
     group graphics
-    onrestart restart zygote
+#    onrestart restart zygote
 
-service zygote /system/bin/app_process -Xzygote /system/bin --zygote --start-system-server
-    class main
-    socket zygote stream 660 root system
-    onrestart write /sys/android_power/request_state wake
-    onrestart write /sys/power/state on
-    onrestart restart media
-    onrestart restart netd
+#service zygote /system/bin/app_process -Xzygote /system/bin --zygote --start-system-server
+#    class main
+#    socket zygote stream 660 root system
+#    onrestart write /sys/android_power/request_state wake
+#    onrestart write /sys/power/state on
+#    onrestart restart media
+#    onrestart restart netd
 
 service drm /system/bin/drmserver
     class main
diff --git a/sp7710ga/prod_sp7710ga.mk b/sp7710ga/prod_sp7710ga.mk
index 41fac37..45e63b5 100644
--- a/sp7710ga/prod_sp7710ga.mk
+++ b/sp7710ga/prod_sp7710ga.mk
@@ -19,7 +19,7 @@ TARGET_BOARD := sp7710ga
 BOARDDIR := device/sprd/$(TARGET_BOARD)
 
 DEVICE_PACKAGE_OVERLAYS := $(BOARDDIR)/overlay
-PRODUCT_PACKAGE_OVERLAYS := vendor/sprd/operator/cmcc/specA/overlay
+#PRODUCT_PACKAGE_OVERLAYS := vendor/sprd/operator/cmcc/specA/overlay
 
 PRODUCT_AAPT_CONFIG := hdpi
 
@@ -68,7 +68,7 @@ PRODUCT_COPY_FILES := \
 	$(BOARDDIR)/ft5x0x_ts.idc:system/usr/idc/ft5x0x_ts.idc \
 	$(BOARDDIR)/headset-keyboard.kl:system/usr/keylayout/headset-keyboard.kl
 
-$(call inherit-product, frameworks/native/build/phone-hdpi-256-dalvik-heap.mk)
+#$(call inherit-product, frameworks/native/build/phone-hdpi-256-dalvik-heap.mk)
 
 # include classified configs
 $(call inherit-product, $(BOARDDIR)/base.mk)
diff --git a/sp7710ga/prod_sp7710gahvga.mk b/sp7710ga/prod_sp7710gahvga.mk
index 54747df..1c4fd03 100644
--- a/sp7710ga/prod_sp7710gahvga.mk
+++ b/sp7710ga/prod_sp7710gahvga.mk
@@ -19,7 +19,7 @@ TARGET_BOARD := sp7710ga
 BOARDDIR := device/sprd/$(TARGET_BOARD)
 
 DEVICE_PACKAGE_OVERLAYS := $(BOARDDIR)/overlay
-PRODUCT_PACKAGE_OVERLAYS := vendor/sprd/operator/cmcc/specA/overlay
+#PRODUCT_PACKAGE_OVERLAYS := vendor/sprd/operator/cmcc/specA/overlay
 
 PRODUCT_AAPT_CONFIG := hdpi
 
@@ -69,7 +69,7 @@ PRODUCT_COPY_FILES := \
 	$(BOARDDIR)/ft5x0x_ts.idc:system/usr/idc/ft5x0x_ts.idc \
 	$(BOARDDIR)/headset-keyboard.kl:system/usr/keylayout/headset-keyboard.kl
 
-$(call inherit-product, frameworks/native/build/phone-hdpi-256-dalvik-heap.mk)
+#$(call inherit-product, frameworks/native/build/phone-hdpi-256-dalvik-heap.mk)
 
 # include classified configs
 $(call inherit-product, $(BOARDDIR)/base.mk)
diff --git a/sp7710ga/prod_sp7710gatri.mk b/sp7710ga/prod_sp7710gatri.mk
index c5b2262..3c02eb2 100644
--- a/sp7710ga/prod_sp7710gatri.mk
+++ b/sp7710ga/prod_sp7710gatri.mk
@@ -19,7 +19,7 @@ TARGET_BOARD := sp7710ga
 BOARDDIR := device/sprd/$(TARGET_BOARD)
 
 DEVICE_PACKAGE_OVERLAYS := $(BOARDDIR)/overlay
-PRODUCT_PACKAGE_OVERLAYS := vendor/sprd/operator/cmcc/specA/overlay
+#PRODUCT_PACKAGE_OVERLAYS := vendor/sprd/operator/cmcc/specA/overlay
 
 PRODUCT_AAPT_CONFIG := hdpi
 
@@ -69,13 +69,13 @@ PRODUCT_COPY_FILES := \
 	$(BOARDDIR)/ft5x0x_ts.idc:system/usr/idc/ft5x0x_ts.idc \
 	$(BOARDDIR)/headset-keyboard.kl:system/usr/keylayout/headset-keyboard.kl
 
-$(call inherit-product, frameworks/native/build/phone-hdpi-256-dalvik-heap.mk)
+#$(call inherit-product, frameworks/native/build/phone-hdpi-256-dalvik-heap.mk)
 
 # include classified configs
 $(call inherit-product, $(BOARDDIR)/base.mk)
 $(call inherit-product, $(BOARDDIR)/proprietories.mk)
 $(call inherit-product, vendor/sprd/operator/cucc/specA/res/boot/boot_res.mk)
-$(call inherit-product, vendor/sprd/UniverseUI/universeui.mk)
+#$(call inherit-product, vendor/sprd/UniverseUI/universeui.mk)
 
 # include standard configs
 $(call inherit-product, $(SRC_TARGET_DIR)/product/full_base.mk)
diff --git a/sp8810ea/AndroidProducts.mk b/sp8810ea/AndroidProducts.mk
index 283a9b5..44d4b22 100644
--- a/sp8810ea/AndroidProducts.mk
+++ b/sp8810ea/AndroidProducts.mk
@@ -16,7 +16,7 @@
 
 PRODUCT_MAKEFILES := $(LOCAL_DIR)/prod_sp8810base.mk
 PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8810plus.mk
-PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8810cmccspecA.mk
-PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8810cmccspecAplus.mk
-PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8810cmccspecAplus_UUI.mk
+#PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8810cmccspecA.mk
+#PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8810cmccspecAplus.mk
+#PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8810cmccspecAplus_UUI.mk
 
diff --git a/sp8825ea/AndroidProducts.mk b/sp8825ea/AndroidProducts.mk
index cbce02a..f4a95cf 100644
--- a/sp8825ea/AndroidProducts.mk
+++ b/sp8825ea/AndroidProducts.mk
@@ -20,4 +20,4 @@ PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8825eacmccspecAplus.mk
 
 PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8825eanativebase.mk
 PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8825eanativeplus.mk
-PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8825eacmccspecAplus_UUI.mk
+#PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8825eacmccspecAplus_UUI.mk
diff --git a/sp8830ea/AndroidProducts.mk b/sp8830ea/AndroidProducts.mk
index 96f7710..dadf864 100644
--- a/sp8830ea/AndroidProducts.mk
+++ b/sp8830ea/AndroidProducts.mk
@@ -15,9 +15,9 @@
 #
 
 
-PRODUCT_MAKEFILES := $(LOCAL_DIR)/prod_sp7735eacuccspecAplus_UUI.mk
-PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8735eaplus_UUI.mk
-PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8835eacmccspecAplus_UUI.mk
+#PRODUCT_MAKEFILES := $(LOCAL_DIR)/prod_sp7735eacuccspecAplus_UUI.mk
+#PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8735eaplus_UUI.mk
+#PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp8835eacmccspecAplus_UUI.mk
 				      
 
 
