diff --git a/sp6820gb/AndroidProducts.mk b/sp6820gb/AndroidProducts.mk
index 4051343..07ac000 100644
--- a/sp6820gb/AndroidProducts.mk
+++ b/sp6820gb/AndroidProducts.mk
@@ -16,5 +16,5 @@
 
 PRODUCT_MAKEFILES := $(LOCAL_DIR)/prod_sp6820base.mk
 PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp6820plus.mk
-PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp6820plusMUI.mk
+#PRODUCT_MAKEFILES += $(LOCAL_DIR)/prod_sp6820plusMUI.mk
 
diff --git a/sp6820gb/base.mk b/sp6820gb/base.mk
index 2672cc2..2efdd38 100644
--- a/sp6820gb/base.mk
+++ b/sp6820gb/base.mk
@@ -7,7 +7,9 @@ BEKEN_FM := fm.$(TARGET_PLATFORM)
 
 RTK_TOOL:= RtkService.apk
 
-PRODUCT_PROPERTY_OVERRIDES :=
+PRODUCT_PROPERTY_OVERRIDES := \
+	ro.moz.omx.hw.max_width=720 \
+	ro.moz.omx.hw.max_height=576
 
 PRODUCT_PACKAGES := \
 	libjni_pinyinime \
@@ -96,7 +98,8 @@ PRODUCT_COPY_FILES := \
 	frameworks/base/data/etc/android.hardware.telephony.gsm.xml:system/etc/permissions/android.hardware.telephony.gsm.xml \
 	frameworks/base/data/etc/android.hardware.camera.autofocus.xml:system/etc/permissions/android.hardware.camera.autofocus.xml \
 	frameworks/base/data/etc/android.hardware.camera.front.xml:system/etc/permissions/android.hardware.camera.front.xml \
-	frameworks/base/data/etc/android.hardware.camera.flash-autofocus.xml:system/etc/permissions/android.hardware.camera.flash-autofocus.xml
+	frameworks/base/data/etc/android.hardware.camera.flash-autofocus.xml:system/etc/permissions/android.hardware.camera.flash-autofocus.xml\
+	$(BOARDDIR)/pixcir_ts.idc:system/usr/idc/pixcir_ts.idc
 
 
 $(call inherit-product, device/sprd/partner/realtek/bt/firmware/device-rtl.mk)
diff --git a/sp6820gb/init.rc b/sp6820gb/init.rc
index 1cdc0cc..8f39ba6 100644
--- a/sp6820gb/init.rc
+++ b/sp6820gb/init.rc
@@ -7,6 +7,8 @@ on early-init
 # create mountpoints
     mkdir /mnt 0775 root system
 
+    import /init.b2g.rc
+
 on init
 
 sysclktz 0
@@ -110,6 +112,9 @@ on fs
 # mount mtd partitions
     # Mount /system rw first to give the filesystem a chance to save a checkpoint
     mount yaffs2 mtd@system /system
+    chmod 0755 /system/b2g/b2g
+    chmod 0755 /system/b2g/updater
+    chmod 0755 /system/b2g/plugin-container
     mount yaffs2 mtd@system /system ro remount
     mount yaffs2 mtd@userdata /data nosuid nodev
     mount yaffs2 mtd@cache /cache nosuid nodev
@@ -395,6 +400,10 @@ service console /system/bin/sh
 
 on property:ro.debuggable=1
     start console
+# add core dump
+    setrlimit 4 -1 -1
+    write /proc/sys/kernel/core_pattern /data/tombstones/core.%e
+    write /proc/sys/fs/suid_dumpable 1
 
 # adbd is controlled via property triggers in init.<platform>.usb.rc
 service adbd /sbin/adbd
@@ -422,7 +431,7 @@ service servicemanager /system/bin/servicemanager
     user system
     group system
     critical
-    onrestart restart zygote
+#    onrestart restart zygote
     onrestart restart media
     onrestart restart surfaceflinger
     onrestart restart drm
@@ -452,15 +461,15 @@ service surfaceflinger /system/bin/surfaceflinger
     class main
     user system
     group graphics
-    onrestart restart zygote
-
-service zygote /system/bin/app_process -Xzygote /system/bin --zygote --start-system-server
-    class main
-    socket zygote stream 666
-    onrestart write /sys/android_power/request_state wake
-    onrestart write /sys/power/state on
-    onrestart restart media
-    onrestart restart netd
+#    onrestart restart zygote
+
+#service zygote /system/bin/app_process -Xzygote /system/bin --zygote --start-system-server
+#    class main
+#    socket zygote stream 666
+#    onrestart write /sys/android_power/request_state wake
+#    onrestart write /sys/power/state on
+#    onrestart restart media
+#   onrestart restart netd
 
 service drm /system/bin/drmserver
     class main
