diff --git a/include/media/AudioSystem.h b/include/media/AudioSystem.h
index 6a15f6e..b974e45 100644
--- a/include/media/AudioSystem.h
+++ b/include/media/AudioSystem.h
@@ -191,6 +191,9 @@ public:
 
     static const sp<IAudioPolicyService>& get_audio_policy_service();
 
+    //fake funciton to avoid gecko reference fail
+    static status_t setFmVolume(float volume);
+
     // ----------------------------------------------------------------------------
 
 private:
@@ -243,6 +246,8 @@ private:
     // list of output descriptors containing cached parameters
     // (sampling rate, framecount, channel count...)
     static DefaultKeyedVector<audio_io_handle_t, OutputDescriptor *> gOutputs;
+
+    static status_t setFmVolumeIndex(int index);
 };
 
 };  // namespace android
diff --git a/include/media/stagefright/MediaSource.h b/include/media/stagefright/MediaSource.h
index 3818e63..09d37db 100644
--- a/include/media/stagefright/MediaSource.h
+++ b/include/media/stagefright/MediaSource.h
@@ -59,6 +59,10 @@ struct MediaSource : public virtual RefBase {
     virtual status_t read(
             MediaBuffer **buffer, const ReadOptions *options = NULL) = 0;
 
+    void SetOMXRunLocally(bool omxRunLocally)
+    {
+        mOMXRunLocally = omxRunLocally;
+    }
     // Options that modify read() behaviour. The default is to
     // a) not request a seek
     // b) not be late, i.e. lateness_us = 0
@@ -111,6 +115,7 @@ struct MediaSource : public virtual RefBase {
 
 protected:
     virtual ~MediaSource();
+    bool mOMXRunLocally;
 
 private:
     MediaSource(const MediaSource &);
diff --git a/media/libmedia/AudioSystem.cpp b/media/libmedia/AudioSystem.cpp
index 5e11586..091a207 100644
--- a/media/libmedia/AudioSystem.cpp
+++ b/media/libmedia/AudioSystem.cpp
@@ -25,6 +25,9 @@
 
 #include <system/audio.h>
 
+#include<unistd.h>
+#include<fcntl.h>
+#define FMFIFO "/data/fmpseudopipe"
 // ----------------------------------------------------------------------------
 
 namespace android {
@@ -680,6 +683,8 @@ status_t AudioSystem::setStreamVolumeIndex(audio_stream_type_t stream, int index
 {
     const sp<IAudioPolicyService>& aps = AudioSystem::get_audio_policy_service();
     if (aps == 0) return PERMISSION_DENIED;
+
+    if (stream == AUDIO_STREAM_FM ) return AudioSystem::setFmVolumeIndex(index);
     return aps->setStreamVolumeIndex(stream, index);
 }
 
@@ -754,6 +759,30 @@ void AudioSystem::clearAudioConfigCache()
     gOutputs.clear();
 }
 
+status_t AudioSystem::setFmVolumeIndex(int index)
+{
+    int result = INVALID_OPERATION;
+
+    int fifo_id = open( FMFIFO ,O_WRONLY|O_NONBLOCK);
+    if(fifo_id != -1) {
+	int buff = index;
+	result = write(fifo_id,&buff,sizeof(int));
+    } else {
+	LOGE("open FM FIFO error\n");
+    }
+    close(fifo_id);
+
+    LOGE("AudioSystem::setFmVolumeIndex index %d result %d \n",index,result);
+    return result;
+}
+
+//fake funciton to avoid gecko reference fail
+status_t AudioSystem::setFmVolume(float volume)
+{
+    LOGE("%s\n volume %f\n",__FUNCTION__,volume);
+    return 0;
+}
+
 // ---------------------------------------------------------------------------
 
 void AudioSystem::AudioPolicyServiceClient::binderDied(const wp<IBinder>& who) {
diff --git a/media/libstagefright/CameraSource.cpp b/media/libstagefright/CameraSource.cpp
old mode 100755
new mode 100644
index 98b2b58..7bc841c
--- a/media/libstagefright/CameraSource.cpp
+++ b/media/libstagefright/CameraSource.cpp
@@ -772,6 +772,7 @@ status_t CameraSource::read(
         (*buffer)->meta_data()->setInt64(kKeyTime, frameTime);
 
 	 //added by jgdu to get physical address @sprd 
+     if (mOMXRunLocally) {
 	 int32_t phy_addr;
 	 ssize_t offset = 0;
     	 size_t size = 0;
@@ -787,6 +788,7 @@ status_t CameraSource::read(
 	 (*buffer)->meta_data()->setInt32(kKeyPlatformPrivate, phy_addr);	
 	 (*buffer)->meta_data()->setInt32(kKeyWidth, mVideoSize.width);
 	 (*buffer)->meta_data()->setInt32(kKeyHeight, mVideoSize.height);	 
+     }
     }
     return OK;
 }
diff --git a/media/libstagefright/MediaSource.cpp b/media/libstagefright/MediaSource.cpp
index fd0e79c..196a752 100644
--- a/media/libstagefright/MediaSource.cpp
+++ b/media/libstagefright/MediaSource.cpp
@@ -18,7 +18,10 @@
 
 namespace android {
 
-MediaSource::MediaSource() {}
+MediaSource::MediaSource()
+{
+    mOMXRunLocally = true;
+}
 
 MediaSource::~MediaSource() {}
 
diff --git a/media/libstagefright/OMXCodec.cpp b/media/libstagefright/OMXCodec.cpp
index aa7fc77..8b4a571 100644
--- a/media/libstagefright/OMXCodec.cpp
+++ b/media/libstagefright/OMXCodec.cpp
@@ -1655,7 +1655,7 @@ OMXCodec::OMXCodec(
     	mDcamColorConverter =
        (void*)dlsym(mHandle, "_ZN7android18DcamColorConverterEiiii");	
     }
-
+    mSource->SetOMXRunLocally(mOMXLivesLocally);
     setComponentRole();
 }
 
