diff --git a/include/system/audio.h b/include/system/audio.h
index 7d80b4d..ec76043 100644
--- a/include/system/audio.h
+++ b/include/system/audio.h
@@ -43,10 +43,10 @@ typedef enum {
     AUDIO_STREAM_ALARM            = 4,
     AUDIO_STREAM_NOTIFICATION     = 5,
     AUDIO_STREAM_BLUETOOTH_SCO    = 6,
-    AUDIO_STREAM_FM               = 7,
+    AUDIO_STREAM_ENFORCED_AUDIBLE = 7, /* Sounds that cannot be muted by user and must be routed to speaker */
     AUDIO_STREAM_DTMF             = 8,
     AUDIO_STREAM_TTS              = 9,
-    AUDIO_STREAM_ENFORCED_AUDIBLE = 10, /* Sounds that cannot be muted by user and must be routed to speaker */
+    AUDIO_STREAM_FM               = 10,
 
     AUDIO_STREAM_CNT,
     AUDIO_STREAM_MAX              = AUDIO_STREAM_CNT - 1,
