diff --git a/drivers/usb/gadget/fastboot.c b/drivers/usb/gadget/fastboot.c
index 28ac9e6..b0c2724 100644
--- a/drivers/usb/gadget/fastboot.c
+++ b/drivers/usb/gadget/fastboot.c
@@ -552,6 +552,8 @@ void cmd_flash(const char *arg, void *data, unsigned sz)
 
 void cmd_erase(const char *arg, void *data, unsigned sz)
 {
+//avoid sp8810/sp8825 can't fastboot erase
+#if 0
 	u8 pnum = 0;
 	unsigned int nblocknum;
 	int pos;
@@ -613,6 +615,7 @@ void cmd_erase(const char *arg, void *data, unsigned sz)
 				return 0;
 			}
 	}
+#endif
 	fastboot_okay("");
 }
 
