diff --git a/apps/system/js/lockscreen.js b/apps/system/js/lockscreen.js
index 21a72de..bf10fa7 100644
--- a/apps/system/js/lockscreen.js
+++ b/apps/system/js/lockscreen.js
@@ -982,7 +982,7 @@ var LockScreen = {
 
 // Bug 836195 - [Homescreen] Dock icons drop down in the UI
 // consistently when using a lockcode and visiting camera
-LockScreen.init();
+//LockScreen.init();
 
-navigator.mozL10n.ready(LockScreen.init.bind(LockScreen));
+//navigator.mozL10n.ready(LockScreen.init.bind(LockScreen));
 
diff --git a/apps/system/js/window_manager.js b/apps/system/js/window_manager.js
index 3d40089..b73ea05 100644
--- a/apps/system/js/window_manager.js
+++ b/apps/system/js/window_manager.js
@@ -956,6 +956,7 @@ var WindowManager = (function() {
   function skipFTU() {
     document.getElementById('screen').classList.remove('ftuStarting');
     InitLogoHandler.animate();
+    LockScreen.init();
     setDisplayedApp(homescreen);
   }

diff --git a/build/settings.py b/build/settings.py
index 6a07e8a..e445209 100644
--- a/build/settings.py
+++ b/build/settings.py
@@ -62,8 +62,8 @@ settings = {
  "lockscreen.passcode-lock.timeout": 0,
  "lockscreen.passcode-lock.enabled": False,
  "lockscreen.notifications-preview.enabled": True,
- "lockscreen.enabled": True,
- "lockscreen.locked": True,
+ "lockscreen.enabled": False,
+ "lockscreen.locked": False,
  "lockscreen.unlock-sound.enabled": False,
  "message.sent-sound.enabled": True,
  "operatorvariant.mcc": "0",
