diff --git a/apps/settings/js/settings.js b/apps/settings/js/settings.js
index b44098c..3996977 100644
--- a/apps/settings/js/settings.js
+++ b/apps/settings/js/settings.js
@@ -400,7 +400,7 @@ var Settings = {
     var key = input.name;
 
     var settings = window.navigator.mozSettings;
-    if (!key || !settings || event.type != 'change')
+    if (!key || !settings || event.type != 'change' ||key=='devtools.debugger.remote-enabled')
       return;
 
    // Not touching <input> with data-setting attribute here
